units/src/amount.rs

```diff
@@ -957,6 +957,15 @@ impl FromStr for Amount {
     fn from_str(s: &str) -> Result<Self, Self::Err> { Amount::from_str_with_denomination(s) }
 }
 
+impl TryFrom<SignedAmount> for Amount {
+    type Error = OutOfRangeError;
+
+    fn try_from(value: SignedAmount) -> Result<Self, Self::Error> {
+        let unsigned_amount = value.to_unsigned()?;
```

Comment:
Just `value.to_unsigned` should be fine. We had a clippy lint for this but it's disabled because of false positives. This one would've been true positive.

Reply:
Done